pkg/terraform/store.go

```diff
@@ -122,6 +123,8 @@ type Setup struct {
 	Scheduler ProviderScheduler
 
 	Meta any
+
+	FrameworkProvider fwprovider.Provider
```

Comment:
Leaving a comment here as a reminder for what we had discussed previously: Because we initialize the framework server and the framework provider at each reconciliation, we may try to extend the provider configuration framework and have the provider directly initialized at each reconciliation in the `ExternalConnecter.Connect` instead of the provider's `terraform.SetupFn` so that we don't have to carry it via the `terraform.Setup`. Let's keep it out of scope for this PR though. We may attempt this "optimization" in a follow-up PR.